### DISCUSSION TITLE
Set conditional value for a relationship picker

### DISCUSSION THREAD

**Discussion by Gabitojr** [Budibase/budibase#7070] [Help] [answered]
Hi.

When selecting the name of the agent: 

![image](https://user-images.githubusercontent.com/108853716/182663317-c413e99c-e31d-45c9-b334-d237364fa3b1.png)

I want it to automatically Update the email field(A relationship picker) with the correspondent email address that belongs to that person.

![image](https://user-images.githubusercontent.com/108853716/182663458-ce10b651-5940-452d-a4d7-93ba1e8af3e7.png)

To do so, I'm using the 'On Change functionalities to update value of that field based on what the end user selects on the Name field.

![image](https://user-images.githubusercontent.com/108853716/182664010-32f4c4cf-eb64-4d45-aec8-c27fe5a96f3d.png)

This is the JS code I'm using which is something very simple:

```
if ($("Form.Fields.Name2") == 'Carlos Garcia'){
	return 'cbgarcia@uamv.edu.ni'
} else  if($("Form.Fields.Name2") == 'Elizabeth Garcia'{
	return 'elizabeth.garcia@uamv.edu.ni'
}
```

However, this doesn't seem to work it is always returning only the 1st email address. Any idea what could I be doing wrong?

**Comment by R2bEEaton** [comment]
Hey! I'm not sure if this is exactly what you want, but you might be able to get away with doing something like this:

If you set the options picker to display the Full Name from a Data Provider, then have the value be their Email Address (ideally stored in the same table), you can have a disabled field update automatically to show the email address to the user, as shown.

![image](https://user-images.githubusercontent.com/34921506/182830523-fe47954c-29c3-4b54-974f-db5b9804e7aa.png)

You can find my export attached to examine it further.
[Options Picker Automatic-export-1659610416626.txt](https://github.com/Budibase/budibase/files/9259057/Options.Picker.Automatic-export-1659610416626.txt)

**Comment by melohagan** [comment]
@Gabitojr Double check your JavaScript: `$("Form.Fields.Name2") == 'Elizabeth Garcia'` - there's a missing closing bracket.

Also is the email address also definitely correct? `elizabeth.garcia@uamv.edu.ni` ?

**Reply by Gabitojr** [reply]
Hi, 

Thank you for answering. Indeed I had an issue with closing bracket (most likely because I tried different methods to accomplish this lol and I forgot to add the bracket at the end )

So right now I have decided to make it more simplistic.

I created  a test app;
![image](https://user-images.githubusercontent.com/108853716/182931160-6674aa8c-0a43-4355-ab19-550a5d082e07.png)

It contains only those two fields within a form

Name - Option Picker
Email - Relationship Picker

I went into the Name field > On change > Define Actions >  and this is What I set up:

![image](https://user-images.githubusercontent.com/108853716/182945788-49c5fbc2-97f3-4042-a805-b136e0b374fd.png)


The JS Code is more simplistic this time (I will add the conditionals later)

![image](https://user-images.githubusercontent.com/108853716/182945991-ba0afc51-51c8-41df-97b1-f55f3392f750.png)

I double checked that the value exists and is valid:

![image](https://user-images.githubusercontent.com/108853716/182946131-c1783adf-5249-4db4-8840-8d3bec9fd617.png)


However, when I go to check the live the application this is what happens:

As soon as I click the name drop down field it shows the names but It doesn't allow me to select anything, nor close it

![image](https://user-images.githubusercontent.com/108853716/182946678-9dcc8a06-04e7-4418-a8ae-cf51e028b552.png)

And I'm getting this error on the console:

![image](https://user-images.githubusercontent.com/108853716/182946520-c7b70369-73de-4464-9bef-6aa4d9b9e102.png)

Any help would be appreciated.

Find attached the app, 

[Text-export-1659645174536.txt](https://github.com/Budibase/budibase/files/9263153/Text-export-1659645174536.txt)

**Comment by melohagan** [comment, marked as answer]
@Gabitojr Thank you very much for the detail.

I was able to replicate the error. The reason is that the relationship picker always expects an array of values. Therefore you can fix your code simply as follows:

```
if( $("Form.Fields.Names") == 'John'){
	return ['cbgarcia@uamv.edu.ni']
} else {
	return ['devagmarketing@uamv.edu.ni']
}
```
**Note** The square brackets have been added to return the value in array format.

With that said this should probably be handled gracefully and not crash your app. I've raised a bug here:
https://github.com/Budibase/budibase/issues/7112

**Reply by Gabitojr** [reply]
Thank you so much!
